src/toolbox/setup/esp8266/windows.ts

```diff
@@ -0,0 +1,92 @@
+import { system, filesystem, print } from 'gluegun'
+import { finished } from 'stream'
+import axios from 'axios'
+import type { GluegunPrint } from 'gluegun'
+import { Extract as ZipExtract } from 'unzip-stream'
+import { promisify } from 'util'
+import { addToPath, setEnv } from '../windows'
+import { INSTALL_DIR } from '../constants'
+
+const finishedPromise = promisify(finished)
+
+const ESP_TOOL = 'https://github.com/igrr/esptool-ck/releases/download/0.4.13/esptool-0.4.13-win32.zip'
+const ESP_TOOL_VERSION = 'esptool-0.4.13-win32'
+const CYGWIN = 'https://www.moddable.com/private/cygwin.win32.zip'
+
+export async function installPython(spinner: ReturnType<GluegunPrint['spin']>) {
+    if (system.which('python') === null) {
+        // For some reason, system.which does not work with winget. This is a workaround for now.
+        let foundWinget = true
+        try {
+            await system.exec('where winget')
+        } catch (error) {
+            foundWinget = false
+        }
```

Comment:
Rather than having this `foundWinget` boolean to use later, I think the `try...catch` conditions provide the necessary logic paths already:

```suggestion
        try {
            await system.exec('where winget')
        } catch (error) {
            print.error('Python is required.')
            print.info('You can download and install Python from python.org/downloads')
            print.info('Or xs-dev can manage installing Python and other dependencies using the Windows Package Manager Client (winget).')
            print.info('You can install winget via the App Installer package in the Microsoft Store.')
            print.info('Please install either Python or winget, then launch a new Command Prompt and re-run this setup.')
            throw new Error("Python is required")
        }
```

Then the `foundWinget` code path can be executed after this `try...catch`.